Failing to build for Android in SPI

Looks like Swift Testing is failing to build for Android on Swift Package Index's site: https://swiftpackageindex.com/builds/875FEA40-0F76-47AF-A7E3-54A39616BA9C

```
/host/spi-builder-workspace/Sources/Testing/SourceAttribution/Backtrace.swift:70:56: error: cannot convert value of type 'UnsafeMutablePointer<UnsafeMutableRawPointer>' to expected argument type 'UnsafeMutablePointer<UnsafeMutableRawPointer?>'
 68 | #elseif os(Android)
 69 |       initializedCount = addresses.withMemoryRebound(to: UnsafeMutableRawPointer.self) { addresses in
 70 |         .init(clamping: backtrace(addresses.baseAddress!, .init(clamping: addresses.count)))
    |                                                        |- error: cannot convert value of type 'UnsafeMutablePointer<UnsafeMutableRawPointer>' to expected argument type 'UnsafeMutablePointer<UnsafeMutableRawPointer?>'
    |                                                        `- note: arguments to generic parameter 'Pointee' ('UnsafeMutableRawPointer' and 'UnsafeMutableRawPointer?') are expected to be equal
 71 |       }
 72 | #elseif os(Linux) || os(FreeBSD)
```

We have special-casing to convert the pointers here to remove optionality because, at one point, the signature of `backtrace()` was annotated such that the pointers were non-optional. This has either changed or is inconsistent across different Android versions.

--- a/Sources/Testing/SourceAttribution/Backtrace.swift
+++ b/Sources/Testing/SourceAttribution/Backtrace.swift
@@ -40,6 +40,16 @@ public struct Backtrace: Sendable {
     self.addresses = addresses.map { Address(UInt(bitPattern: $0)) }
   }
 
+#if os(Android) && !SWT_NO_DYNAMIC_LINKING
+  /// The `backtrace()` function.
+  ///
+  /// This function was added to Android with API level 33, which is higher than
+  /// our minimum deployment target, so we look it up dynamically at runtime.
+  private static let _backtrace = symbol(named: "backtrace").map {
+    castCFunction(at: $0, to: (@convention(c) (UnsafeMutablePointer<UnsafeMutableRawPointer?>, CInt) -> CInt).self)
+  }
+#endif
+
   /// Get the current backtrace.
   ///
   /// - Parameters:
@@ -66,9 +76,9 @@ public struct Backtrace: Sendable {
         initializedCount = .init(clamping: backtrace(addresses.baseAddress!, .init(clamping: addresses.count)))
       }
 #elseif os(Android)
-      initializedCount = addresses.withMemoryRebound(to: UnsafeMutableRawPointer.self) { addresses in
-        .init(clamping: backtrace(addresses.baseAddress!, .init(clamping: addresses.count)))
-      }
+#if !SWT_NO_DYNAMIC_LINKING
+      initializedCount = .init(clamping: _backtrace?(addresses.baseAddress!, .init(clamping: addresses.count)))
+#endif
 #elseif os(Linux) || os(FreeBSD) || os(OpenBSD)
       initializedCount = .init(clamping: backtrace(addresses.baseAddress!, .init(clamping: addresses.count)))
 #elseif os(Windows)

